### ISSUE TITLE
Multiple Launches created during Parallel Execution.

### ISSUE THREAD

**Issue by sudarsunperu** [Invictum/serenity-reportportal-integration#104]
Hi @Invictum , 
I'm executing my Tests using serenity Batches. 
I've configured my job to run all my Test Cases with 5 batches. I'm observing that for 1 run, it creates 5 **Launches** on Report Portal (equal to the number of batches). However I would like to configure it as 1 execution (5 batches) creates 1 **Launch** . 
If there is a way to pass a **Launch** to all the batches and they update them, it would be great. 

Can you please help with configuring it as per the above expectation?

**Comment by Invictum** [comment]
Hi @sudarsunperu 

Unfortunately integration doesn't support batch run merge, but potentially you can try to utilize [Modules merge launches](https://github.com/Invictum/serenity-reportportal-integration#merge-launches) feature to achieve it (I didn't try it). Configuration will depend on you jobs structure and rules, but in general you should provide two options:
- `erenity.rp.modules.count` set it to number of batches - 5 in your case
- `erenity.rp.communication.dir` path to directory that will be available for all batch jobs. If you run batches in scope of 1 job, just make sure that directory isn't removed between batches. If you run batch jobs in parallel you may try to specify some network directory. Use it with caution because all the data inside that directory will be removed at the end of execution.

**Comment by sudarsunperu** [comment]
Hi @Invictum 

Thank you for your response. I will checkout the solution that you have proposed. Alternately I had an approach in mind and would like to know your thoughts on the same. 

Would it be possible to pass a **Launch id** to all the batches, and each batch can update the same launch instead of creating a new one. So before starting the batches, I can create a Launch using the Reportportal API and pass it on to all the batches and start them.

Maybe we can configure it in the `@BeforeClass` in the Runner using `ReportIntegrationConfig` (just an idea not sure of the feasibility).

**Comment by sudarsunperu** [comment]
Hi @Invictum 
I passed the below properties and found couple of files created in the `serenity.rp.communication.dir`. 

- `serenity.rp.communication.dir`="sync-dir"
- `serenity.rp.modules.count`=2

However in Report portal it is still showing up as two separate Launches even after the Tests complete. And I'm able to see the sync-dir created with two files in it (0 Bytes).
![image](https://user-images.githubusercontent.com/18679720/84343117-92436580-aba7-11ea-9995-5d9c82974603.png)

The documentation says 
```With merge feature activation each submodule still produce separate launch on execution phase, but they will be merged into one at the end of all tests execution.```

But it doesn't merge the Launches. Can you please help?

**Comment by Invictum** [comment]
@sudarsunperu 
Seems like you found an issue. Nice catch! I will create a separate ticket to track it https://github.com/Invictum/serenity-reportportal-integration/issues/107

Current issue will be closed
